### PR TITLE
Refactor requirements form to be more modular

### PR DESCRIPTION
**Pull request by StevePole**
Some tests failing…

### CODE REVIEW

File: angular/components/tw-dynamic-form-control/tw-dynamic-form-control.directive.js [hunk 1]
```diff
@@ -132,14 +133,14 @@
 				<div ng-switch-when='upload'> \
 					<tw-upload \
 						name='{{$ctrl.name}}' \
-						description='{{$ctrl.placeholder}}' \
```
Comment (lapostoj) [inline angular/components/tw-dynamic-form-control/tw-dynamic-form-control.directive.js]: Was it use nowehre else than our usecases?

File: angular/components/tw-requirements-form/tw-requirements.service.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+(function(angular) {
+	'use strict';
+
+	angular
+		.module('tw.form-components')
+		.service('TwRequirementsService', TwRequirementsService);
```
Comment (lapostoj) [inline angular/components/tw-requirements-form/tw-requirements.service.js]: Definitely not against it, but to understand: what pushed you to put that part in a service?

Comment (StevePole) [inline angular/components/tw-requirements-form/tw-requirements.service.js]: Wanted to be able to reuse some of this 'generic' requirements logic across components.

File: angular/components/tw-dynamic-form-control/tw-dynamic-form-control.directive.js (hunk 1 above)
Comment (StevePole) [inline angular/components/tw-dynamic-form-control/tw-dynamic-form-control.directive.js]: I've kept it backwards compatible, you can still use description but you can also use placeholder, so I switched this use over to the new value.

**Comment by StevePole** [comment]
Yeah, the commit is in there:
image

**Pull request merged**